be/src/vec/functions/match.h

```diff
@@ -0,0 +1,211 @@
+#include <string>
+#include <utility>
+
+#include "common/config.h"
+#include "common/consts.h"
+#include "common/logging.h"
+#include "common/status.h"
+#include "olap/inverted_index_parser.h"
+#include "vec/aggregate_functions/aggregate_function.h"
+#include "vec/columns/column.h"
+#include "vec/core/block.h"
+#include "vec/core/column_numbers.h"
+#include "vec/core/column_with_type_and_name.h"
+#include "vec/core/types.h"
+#include "vec/data_types/data_type_number.h"
+#include "vec/exprs/vmatch_predicate.h"
+#include "vec/functions/function.h"
+#include "vec/functions/simple_function_factory.h"
+
+namespace doris {
+class FunctionContext;
+} // namespace doris
+
+namespace doris::vectorized {
+
+struct InvertedIndexCtx;
+
+class FunctionMatchBase : public IFunction {
+public:
+    size_t get_number_of_arguments() const override { return 2; }
+
+    String get_name() const override { return "match"; }
+
+    /// Get result types by argument types. If the function does not apply to these arguments, throw an exception.
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return std::make_shared<DataTypeUInt8>();
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override;
+
+    virtual Status execute_match(const std::string& column_name,
+                        const std::string& match_query_str,
+                        size_t input_rows_count,
+                        const ColumnString* query_values,
+                        doris::InvertedIndexParserType parser_type,
+                        ColumnUInt8::Container& result) = 0;
+
+protected:
+    bool is_equal(std::set<std::string>& v1, std::set<std::string>& v2);
+};
+
+class FunctionMatchAny : public FunctionMatchBase {
+public:
+    static constexpr auto name = "match_any";
+    static FunctionPtr create() { return std::make_shared<FunctionMatchAny>(); }
+
+    String get_name() const override { return name; }
+
+    virtual Status execute_match(const std::string& column_name,
+                        const std::string& match_query_str,
+                        size_t input_rows_count,
+                        const ColumnString* query_values,
+                        doris::InvertedIndexParserType parser_type,
+                        ColumnUInt8::Container& result) override;
+};
+
+class FunctionMatchAll : public FunctionMatchBase {
+public:
+    static constexpr auto name = "match_all";
+    static FunctionPtr create() { return std::make_shared<FunctionMatchAll>(); }
+
+    String get_name() const override { return name; }
+
+    virtual Status execute_match(const std::string& column_name,
+                        const std::string& match_query_str,
+                        size_t input_rows_count,
+                        const ColumnString* query_values,
+                        doris::InvertedIndexParserType parser_type,
```

Comment:
we need more params in this function, like parser_granularity

Reply:
I will use `InvertedIndexCtx` instead of `InvertedIndexParserType`, can get other parameters from 
 the `InvertedIndexCtx`